include/picongpu/plugins/saxs/Saxs.hpp

```diff
@@ -0,0 +1,813 @@
+                DataSpace< DIM1 > bufferSize;
+                auto totalNumVectors = numVectors.productOfComponents( );
+                if ( outputLayout == OutputMemoryLayout::MirrorOnNodes )
+                {
+                    // All vectors are stored on every node.
+                    bufferSize = DataSpace< DIM1 >( totalNumVectors );
+                    // Initiate the additional amplitude storage for the reduce
+                    // operation and initiate it with zeros.
+                    amplitudeMaster.assign( totalNumVectors, complex_X( 0.0 ) );
+                    bufferSize_i = totalNumVectors;
+                }
+                else
+                {
+                    countRanks = gc.getGpuNodes( ).productOfComponents( );
+                    // Number of scattering vectors in all but last chunk.
+                    // (ceil integer division)
+                    bufferSize_i = totalNumVectors / countRanks +
+                        ( ( totalNumVectors % countRanks ) != 0 );
+                    // Number of scattering vectors on the last chunk.
+                    resOfVectors = bufferSize_i - ( bufferSize_i * countRanks -
+                        totalNumVectors);
+                    // Initiate the additional amplitude storage for receiving
+                    // data and initiate it with zeros.
+                    amplitudeReceive.assign(
+                        bufferSize_i,
+                        complex_X( 0.0 )
+                    );
+                    bufferSize = DataSpace< DIM1 >( bufferSize_i );
+                }
+                // Allocate amplitude buffer.
+                amplitude = pmacc::memory::makeUnique< ComplexBuffer >(
+                    bufferSize );
+                // Initialize, on device, its fields with zero.
+                amplitude->getDeviceBuffer( ).setValue( 0.0 );
+
+                // Set the q-space grid spacing.
+                q_step = ( q_max - q_min ) /
+                         precisionCast< float_X >( numVectors );
+
+                // Rank 0 creates the output directory.
+                pmacc::Filesystem< simDim > & fs =
+                    Environment< simDim >::get( ).Filesystem( );
+                if ( isMaster )
+                {
+                    fs.createDirectory( "saxsOutput" );
+                    fs.setDirectoryPermissions( "saxsOutput" );
+                }
+
+                // Chose the solver for populating a TmpField with the form factor
+                // density of the particles.
+                using FormFactorSolver =
+                    typename traits::DetermineFormFactorSolver<
+                    T_ParticlesType >::type;
+                // Output unit:
+                const float_64 amplitudeUnit = static_cast< float_64 >(
+                    FieldTmp::getUnit< FormFactorSolver >( )[ 0 ] ) *
+                    CELL_VOLUME * UNIT_LENGTH * UNIT_LENGTH * UNIT_LENGTH
+                    * ELECTRON_RADIUS_SI;
```

Comment:
`CELL_VOLUME * UNIT_LENGTH * UNIT_LENGTH * UNIT_LENGTH`
 should be equivalent to 
`CELL_WIDTH_SI * CELL_HEIGHT_SI * CELL_DEPTH_SI` 
but I agree that the second way of writing it may be more understandable.